pkg/module/disk_ref.go

```diff
@@ -61,7 +79,7 @@ func (d *diskRef) Read() (*storage.Version, error) {
 	}
 	ver.Mod = mod
 
-	sourceFile, err := d.fs.Open(filepath.Join(d.root, fmt.Sprintf("%s.zip", d.version)))
+	sourceFile, err := d.fs.Open(filepath.Join(packagePath, fmt.Sprintf("%s.zip", d.version)))
```

Comment:
In fact, could you make `ver.Zip` below take `&refCloser{zip: sourceFile, ref: d}` and then you can in this file do the following: 

```golang
type refCloser struct {
	zip  io.ReadCloser
	ref Ref
}

func (rc *refCloser) Close() error {
	rc.zip.Close()
	return rc.Ref.Clear()
}

func (rc *refCloser) Read(p []byte) (n int, err error) {
	return rc.zip.Read(p)
}
```

This way we ensure a path is cleaned up once its Zip is fully closed. 

This can be another pr since it addresses https://github.com/gomods/athens/issues/355 but it's also very closely related to what this pr does. so it's up to you :)

Reply:
absolutely! I'll make that change

Reply:
Actually I've commented on the issue for some clarification when you get a chance.